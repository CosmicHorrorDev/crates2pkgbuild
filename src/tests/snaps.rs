use super::utils::{init_log, sample_package, FullFakeEnv, SampleCrate};

macro_rules! gen_snapshot_tests {
    ( $( ( $test_name:ident, $sample:expr ) ),* $(,)? ) => {
        $(
            #[test]
            fn $test_name() {
                init_log();

                let env = FullFakeEnv::new();
                let crate_ = sample_package($sample);
                let pkgbuild = crate::fetch_with_env(crate_.clone(), &env).unwrap();
                insta::with_settings!({ info => &crate_ }, {
                    insta::assert_snapshot!(pkgbuild);
                });
            }
        )*
    }
}

gen_snapshot_tests!(
    (cargo_pgo, SampleCrate::CargoPgo),
    (inlyne, SampleCrate::Inlyne),
    (ra_multiplex, SampleCrate::RaMultiplex),
    (regex_cli, SampleCrate::RegexCli),
    (to_html, SampleCrate::ToHtml),
);
