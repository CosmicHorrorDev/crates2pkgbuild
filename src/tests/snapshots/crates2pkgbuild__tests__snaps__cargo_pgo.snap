---
source: src/tests/snaps.rs
expression: pkgbuild
info:
  name: cargo-pgo
  val:
    maintainers:
      - Test <dummy@web.site>
    contributors: []
    version: 0.2.8
    release-version: 1
    arch:
      - x86_64
    deps:
      - gcc-libs
    make-deps:
      - cargo
    env:
      CARGO_TARGET_DIR:
        value: target
        steps:
          - build
          - check
      RUSTUP_TOOLCHAIN:
        value: stable
        steps:
          - build
          - check
          - prepare
    test-in-check: false
    missing-license-files: true
    find-build-completions: []
    static-completions-within: ~
---
# Maintainer: Test <dummy@web.site>

pkgname=cargo-pgo
pkgver=0.2.8
pkgrel=1
arch=(x86_64)
license=(MIT)
pkgdesc='Cargo subcommand for optimizing Rust binaries with PGO and BOLT.'
url='https://github.com/kobzol/cargo-pgo'
depends=(gcc-libs)
makedepends=(cargo)
source=("$pkgname-$pkgver.tar.gz::https://static.crates.io/crates/$pkgname/$pkgname-$pkgver.crate")
# Upstream checksum. Can be verified with
# $ curl https://index.crates.io/ca/rg/cargo-pgo | grep -F '"0.2.8"' | jq .cksum
sha256sums=(2844634a2baa76c7ac0335a5a3a071c7143c70fe1553e45f37d72ac623b0ddd2)

prepare() {
  cd "$srcdir/$pkgname-$pkgver"

  export RUSTUP_TOOLCHAIN=stable
  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
}

build() {
  cd "$srcdir/$pkgname-$pkgver"

  export CARGO_TARGET_DIR=target
  export RUSTUP_TOOLCHAIN=stable
  cargo build --frozen --release
}

package() {
  cd "$srcdir/$pkgname-$pkgver"

  install -Dm755 target/release/cargo-pgo "$pkgdir/usr/bin/cargo-pgo"
  install -Dm644 README.md "$pkgdir/usr/share/doc/$pkgname/README.md"
}

# vi: filetype=sh shiftwidth=2 expandtab
